types/react/index.d.ts

```diff
@@ -129,6 +127,8 @@ declare namespace React {
 
     type ComponentState = any;
 
+    type Key = string | number;
+
```

Comment:
Huh, moving this around fixed it for me locally, but not yet on CI. Unsure what to do about this.

The issue is that [this test](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ce830707266deab17289b07b57df0e37e03e04ed/types/react/test/index.ts#L558) gets back `ReactText[]` instead of `Key[]`. The definitions of both are identical, so it seems kinda arbitrary:

```ts
type Key = string | number;
type ReactText = string | number;
```

cc @andrewbranch

Reply:
`$ExpectType` is somewhat fragile because I think it tests for exact string match, I would maybe change the test so that the `$ExpectType` is `true` or something, by wrapping it in a conditional type.